cypress-io/cypress
# feat: add v8 snapshot to monorepo

- Closes #22980 

### User facing changelog


n/a

### Additional details


As part of the v8 snapshots feature, we need access to a new package called v8-snapshot which was created previously partly to support the overall functionality. In order to utilize it in the Cypress app, I want to move v8-snapshot from being a [standalone project](https://github.com/cypress-io/v8-snapshot) to the monorepo. The code has mostly stayed the same as what written originally. I just cleaned up a few things and standardized it to be in line with the monorepo.

### Steps to test


For now this package is completely self contained. Future PRs will incorporate this into V8 snapshots. I would recommend checking out the `ReadMe` and documentation in the code to follow what is happening with it. You can also check out some of the [integration tests](https://github.com/cypress-io/cypress/blob/ryanm/feat/add-v8-snapshot-to-monorepo/tooling/v8-snapshot/test/integration/loading.spec.ts) which show how it would be utilized for a given project.

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
